source/extensions/filters/network/connection_limit/connection_limit.h

```diff
@@ -0,0 +1,96 @@
+#pragma once
+
+#include <chrono>
+#include <thread>
+
+#include "envoy/event/dispatcher.h"
+#include "envoy/event/timer.h"
+#include "envoy/extensions/filters/network/connection_limit/v3/connection_limit.pb.h"
+#include "envoy/network/filter.h"
+#include "envoy/runtime/runtime.h"
+#include "envoy/stats/stats_macros.h"
+
+#include "common/common/thread_synchronizer.h"
+#include "common/runtime/runtime_protos.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace NetworkFilters {
+namespace ConnectionLimitFilter {
+
+/**
+ * All connection limit stats. @see stats_macros.h
+ */
+#define ALL_CONNECTION_LIMIT_STATS(COUNTER, GAUGE)                                                 \
+  COUNTER(limited_connections)                                                                     \
+  GAUGE(active_connections, Accumulate)
+
+/**
+ * Struct definition for connection limit stats. @see stats_macros.h
+ */
+struct ConnectionLimitStats {
+  ALL_CONNECTION_LIMIT_STATS(GENERATE_COUNTER_STRUCT, GENERATE_GAUGE_STRUCT)
+};
+
+/**
+ * Configuration shared across all connections. Must be thread safe.
+ */
+class Config : Logger::Loggable<Logger::Id::filter> {
```

Comment:
across all connections?  I thought from docs it was "on a filter chain basis".
If something is shared across connections for a given listener, it doesn't need to be thread safe, because while there are many connections only one may be active at a time.

Reply:
Yes, you're right, updated.